### PR TITLE
Reader/update discover byline

### LINKED ISSUE #10870: Reader Discover: Update stream card byline
Update discover stream card bylines with:
screen shot 2017-01-23 at 1 55 22 pm

1. Source site blavatar
2. Source pick author ( if available )
3. Source site
4. Source site follow button

Displaying "Featured on Discover" on all streams is TBD

### PR DESCRIPTION
**Pull request by jhnstn**
Fixes #10870 
This updates the byline for discover cards in the ~~reader~~ discover stream. 
Currently the byline displays the editor who made the discover pick.  This PR changes the byline for site and post picks where we have either a `site_id` or a `post_id`. It uses this information, along with the discover pick post to render the byline.

This PR does not handle posts from external sites. For now, these external posts still display the editor byline.

**Site Pick:**
screen shot 2017-02-01 at 3 57 03 pm

**Non-Site Pick:**
screen shot 2017-02-01 at 3 57 35 pm

_**note**_: The date and primary tag are pulled from the discover post.

**Why**
These changes are part of a larger [discover tab update](https://github.com/Automattic/wp-calypso/projects/17). 

**Key changes**
- Change`originalPost` to `discoverPick`. The new prop now holds the pick site and post.
- Loads the discover pick site into the redux state tree

**UPDATE**
I removed the "Featured on Discover" from this PR and pulled the scope back so these changes only appear on the discover tab

### CODE REVIEW

**Comment by fraying** [comment]
Hi @jhnstn! Looks good. But one thing: we added the separate follow line ("Follow Site Name," below the excerpt) because the byline belonged to the Discover author. If we're changing it to the linked site, then the follow link should go in the standard top, right position, like so:

fixed

**Comment by jhnstn** [comment]
@fraying yup that's up next in #11123. I wanted to limit the scope of this PR to the byline.
I do have most of the code ready to make that change so it will be a quick turn around

**Comment by fraying** [comment]
@jhnstn Woot! Rock on, then.

File: client/blocks/reader-post-card/style.scss [hunk 1]
```diff
@@ -700,3 +700,19 @@ $reader-post-card-breakpoint-small: "( max-width: 550px )";
 .is-section-devdocs .reader-post-card__featured-image {
 	bottom: 16px;
 }
+
+.reader-post-card__discover-post-flag {
```
Comment (jhnstn) [inline client/blocks/reader-post-card/style.scss]: The styling here is the same as the [recommended posts](https://github.com/Automattic/wp-calypso/blob/master/client/reader/stream/_style.scss#L417) but positioning is different. I opted to duplicate the rules here but am open to DRY'n it up.

**Comment by jhnstn** [comment]
@kjellr,  this will add "Featured on Discover" flag to discover picks. I added that to this PR since we will be losing the discover editor byline for these cards in non-discover streams. I used the same styling as the Recommended posts:

screen shot 2017-02-03 at 1 27 20 pm

How's this look ?

**Comment by fraying** [comment]
What problem is this solving?

File: client/blocks/reader-post-card/index.jsx [hunk 2]
```diff
@@ -158,15 +161,33 @@ export default class ReaderPostCard extends React.Component {
 				</StandardPost>;
 		}
 
+		// set up post byline
+		let postByline;
+
+		if ( isDiscoverStream && ! isEmpty( discoverPick ) ) {
+			// create a post like object with some props from the discover post
```
Comment (samouri) [inline client/blocks/reader-post-card/index.jsx]: This seems like it should be a post-normalizer rule: `rule-post-to-open`.

Then we probably wouldn't need any special cases in this file!

File: client/blocks/reader-post-card/index.jsx [hunk 3]
```diff
@@ -48,9 +51,8 @@ export default class ReaderPostCard extends React.Component {
 	};
 
 	propagateCardClick = () => {
-		// If we have an original post available (e.g. for a Discover pick), send the original post
-		// to the full post view
-		const postToOpen = this.props.originalPost ? this.props.originalPost : this.props.post;
+		// If we have an discover pick post available, send the discover pick to the full post view
+		const postToOpen = get( this.props, 'discoverPick.post' ) || this.props.post;
```
Comment (jhnstn) [inline client/blocks/reader-post-card/index.jsx]: `post.discoverPick` is initially `undefined` till we load the discover pick post or the discover site post from the stores/redux.

File: client/blocks/reader-post-card/index.jsx (hunk 2 above)
Comment (jhnstn) [inline client/blocks/reader-post-card/index.jsx]: Sounds like a plan!
This PR is blocking a few next steps so I'd like to ship it soon. I'm all for normalizing this but would like to do that in a new PR.

File: client/reader/route/index.js [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
 const SITE_URL_BASE = '/read/blogs/';
 
-const PRETTY_FEED_URLS = {
-	12733228: '/discover'
-};
+const DISCOVER_SITE_ID = config( 'discover_blog_id' );
+const DISCOVER_FEED_ID = config( 'discover_feed_id' );
+
+const prettyFeedUrls = {};
+prettyFeedUrls[ DISCOVER_FEED_ID ] = '/discover';
```
Comment (samouri) [inline client/reader/route/index.js]: shouldn't stop us from landing this but with es6 we can:
```jsx 
const prettyFeedUrls = {
  [ DISCOVER_FEED_ID ]: '/dicover'
}
```

**Comment by samouri** [comment]
@jhnstn: I think we are accidentally stripping the site from the byline for non-discover streams.
After vs. Before:
https://cldup.com/GU6YFTBLwD.png

**Comment by samouri** [comment]
I think if we just set the default for `shouldShowSiteNameOnCards` to true that will fix things

**Comment by jhnstn** [comment]
@samouri yeah that would do it. I just moved the default prop up to post card index.

**Comment by samouri** [comment]
Code LGTM. Testing!

**Comment by samouri** [comment]
Functionality LGTM, thanks for fixing it up @jhnstn!

**Pull request merged**